sinks/cortex/cortex.go

```diff
@@ -0,0 +1,226 @@
+package cortex
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/snappy"
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/sirupsen/logrus"
+	"github.com/stripe/veneur/v14"
+	"github.com/stripe/veneur/v14/samplers"
+	"github.com/stripe/veneur/v14/sinks"
+	"github.com/stripe/veneur/v14/ssf"
+	"github.com/stripe/veneur/v14/trace"
+	"github.com/stripe/veneur/v14/util"
+)
+
+const (
+	// MaxConns is the number of simultaneous connections we allow to one host
+	MaxConns = 100
+	// DefaultRemoteTimeout after which a write request will time out
+	DefaultRemoteTimeout = time.Duration(30 * time.Second)
+)
+
+// CortexMetricSink writes metrics to one or more configured Cortex instances
+// using the prometheus remote-write API. For specifications, see
+// https://github.com/prometheus/compliance/tree/main/remote_write
+type CortexMetricSink struct {
+	URL           string
+	RemoteTimeout time.Duration
+	ProxyURL      string
+	client        *http.Client
+}
+
+type CortexMetricSinkConfig struct {
+	URL           string        `yaml:"url"`
+	RemoteTimeout time.Duration `yaml:"remote_timeout"`
+	ProxyURL      string        `yaml:"proxy_url"`
```

Comment:
If you would like to, we can make the type of this a custom `util.yamlUrl` (or something like that), which contains url.URL and implements `UnmarshalYAML`. That way, we parse the input when the config is loaded (similar to how `time.Duration` works, but `go-yaml` supports `time.Duration` natively).

Reply:
skipping this for now as discussed offline